### PR TITLE
chore: upgrade to typedoc 0.10.0

### PR DESCRIPTION
**Pull request by raymondfeng**
### Description

Upgrade to typedoc 0.10.0.

#### Related issues

- connect to <link_to_referenced_issue>

### Checklist

- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style
  guide](http://loopback.io/doc/en/contrib/style-guide.html)

### CODE REVIEW

**Comment by raymondfeng** [comment]
@virkt25 I added a commit to significantly improved the performance of `typedoc` parsing. No more timeouts. 
PTAL.

File: test/ts-parser.test.js [hunk 1]
```diff
@@ -5,10 +5,12 @@ var expect = require('chai').expect;
 var TSParser = require('../lib/tsParser');
 
 describe('TypeScript Parser Test', function() {
-  var tsOptions = {tsconfig: path.join(__dirname, 'tsconfig.json')};
+  var tsOptions = {
+    excludeNotExported: false,
```
Comment (virkt25) [inline test/ts-parser.test.js]: Any reason this has to be set to false for tests? `fixtures/ts/Greeter*.ts` seem to be exporting the classes.

Comment (raymondfeng) [reply test/ts-parser.test.js]: @virkt25 Great catch! That led to https://github.com/TypeStrong/typedoc/pull/694.

File: package.json [hunk 2]
```diff
@@ -10,7 +10,7 @@
     "url": "git://github.com/strongloop/strong-docs.git"
   },
   "scripts": {
-    "test": "mocha"
+    "test": "mocha --timeout 15000"
```
Comment (virkt25) [inline package.json]: This is causing failure on Jenkins (Mac). I've run into the issue before on other repos (each run was different and I ended up having to disable timeout completely `this.timeout(0)`. The machine is really slow but maybe we can bump this up for CI ... or just trigger it again till it passes.

File: test/ts-parser.test.js [hunk 3]
```diff
@@ -5,24 +5,43 @@ var expect = require('chai').expect;
...
+      excludeNotExported: false,
+      tsconfig,
+    });
     var parsedData = tsParser.parse();
     expect(parsedData.sections).to.have.length(3);
     expect(parsedData.constructs).to.have.length(1);
     expect(parsedData.errors).to.have.length(0);
   });
 
+  it.skip('should parse TS file for exported ' +
```
Comment (bajtos) [inline test/ts-parser.test.js]: "should parse TS file for exported" - looks like an incomplete sentence to me, for exported _what_?

It took me a while to figure out why this test is marked as `skipped`, I assume it requires a feature that's not supported by typedoc yet. Is that correct? In which case, please add a regular code comment explaining the situation and pointing to the typedoc issue.

```js
// Typedoc does not support this feature yet
// See https://github.com/TypeStrong/typedoc/pull/694
it.skip('should parse TS file for exported <what> - not supported by typedoc yet', function() {
  // ...
});
```

File: test/ts-parser.test.js [hunk 4]
```diff
@@ -31,10 +50,12 @@ describe('TypeScript Parser Test', function() {
   });
 
   it('should parse Array.includes with es2016', function() {
-    this.timeout(90000);
     var file = path.join(__dirname, 'fixtures/ts/Greeter.es2016.ts');
     var tsFiles = [file];
-    var tsParser = new TSParser(tsFiles, {target: 'es2016'});
+    var tsParser = new TSParser(tsFiles, {
+      target: 'es2016',
```
Comment (bajtos) [inline test/ts-parser.test.js]: IIRC, ES6=ES2015, ES2016 added only few features, the next big revision is ES2017 with async/await. Did you perhaps mean `es2015` as the taget?

I am surprised the target actually matters.

Anyhow, I don't have much opinions about this, just found a suspicious string. Please use whatever value makes most sense to you and works well.

**Pull request merged**